coreclient/src/clients/mod.rs

```diff
@@ -523,13 +523,18 @@ impl CoreUser {
                 content.clone(),
             );
             conversation_message.store(&transaction, &mut notifier)?;
+
+            // Notify as early as possible to react to the not yet sent message
+            notifier.notify();
+
             let mut group = Group::load(&transaction, group_id)?
                 .ok_or(anyhow!("Can't find group with id {group_id:?}"))?;
             let params = group.create_message(&transaction, content)?;
             // Immediately write the group back. No need to wait for the DS to
             // confirm as this is just an application message.
             group.store_update(&transaction)?;
             // Also, mark the message (and all messages preceeding it) as read.
+            let mut notifier = self.store_notifier();
```

Comment:
Is this needed?

Reply:
Yes, the previous notifier is gone (L 528) as `notify` consumes it. If we are offline, we fail in L 530 and the notification is never sent out.

When we split this operation in a state machine with clearly separate steps this will become more obvious.

Reply:
I see, I didn't realize it would be consumed.